editoast/src/config.rs

```diff
@@ -0,0 +1,29 @@
+use std::collections::HashMap;
+
+use serde::Deserialize;
+
+#[derive(Deserialize, PartialEq, Debug)]
+pub struct Config {
+    pub search: HashMap<String, Search>,
+}
+
+#[derive(Deserialize, PartialEq, Debug)]
+pub struct Search {
+    pub table: String,
+    pub columns: Vec<String>,
+    pub result: SearchResult,
+}
+
+#[derive(Deserialize, PartialEq, Debug)]
+pub struct SearchResult {
+    pub joins: Option<String>,
+    pub columns: HashMap<String, String>,
+}
+
+pub fn config() -> &'static Config {
+    lazy_static! {
```

Comment:
The use of `lazy_static` is overkill here. You can create a `Config` at server startup then store it as a `State`. Take the example on `MapLayers` which is very similar on what you want to do.